pkg/expect/expect_test.go

```diff
@@ -18,12 +18,25 @@ package expect
 
 import (
 	"os"
+	"os/exec"
 	"testing"
 	"time"
 )
 
+func findBinaryOrSkip(t *testing.T, binName string) string {
+	t.Helper()
+
+	if binPath, _ := exec.LookPath(binName); binPath != "" {
+		return binPath
+	}
+
+	t.Skipf("Binary (%q) does not exist", binName)
```

Comment:
I don't know those tests very well, however I'm worried that we just decide to skip tests based on external dependency. This could lead to bad situation long term as we don't notice changes in CI base image. I think the tests should fail and decide if some tests should be removed as they require non standard binary.

Reply:
Yes, it's a good point. But I don't think we should remove any of them, because the purpose of the tests in `expect_test.go` is just to verify the expect's functionality, and obviously they are useful. 

The issue here is that NixOS is really special, which has different package management tool & configuration. It means some tools/binaries may locate on different places. 

Note that [NewExpect](https://github.com/etcd-io/etcd/blob/a1f24af310bb1e11616985bf75b98384aa2c8c42/pkg/expect/expect_test.go#L26) internally calls `exec.Command`, and which actually depends on [exec.LookPath](https://github.com/golang/go/blob/3efc7215cbf6c7842cba0f5ebe90f72f0b6de9e1/src/os/exec/exec.go#L175) as well if the command name contains no path separators. If we decide to support the special case like NixOS, then a simpler solution is just to replace the command name with its base name, such as replacing "/bin/echo" with "echo".

Reply:
Good catch, simpler solution would be to just remove absolute paths.

Reply:
Oh awesome 👍

Reply:
Updated